### PR TITLE
ignore STOP_WAITINGs, derive lower bound for packets to include in ACK

### LINKED ISSUE #964: ignore STOP_WAITING frames
We still need to use them for gQUIC for now, but we should remove them for IETF QUIC (and derive the lower bound for acking packets ourselves).

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #964.

This PR implements the logic to derive the lower bound of which packets to include in an ACK frame. It does so by keeping track of ACKs for ACKs. This will be needed when STOP_WAITING frames are removed (which will happen in IETF QUIC, and a future PR will already make that change for `VersionTLS`).
We're now still sending STOP_WAITING frames, since Chrome will need them. However, we're only parsing and then ignoring any STOP_WAITING that we receive.

### CODE REVIEW

File: ackhandler/interfaces.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ type SentPacketHandler interface {
 
 	SendingAllowed() bool
 	GetStopWaitingFrame(force bool) *wire.StopWaitingFrame
+	GetLowestPacketNotConfirmedAcked() protocol.PacketNumber
```
Comment (marten-seemann) [inline ackhandler/interfaces.go]: @lucas-clemente: Any idea for a better name?

File: Changelog.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## v0.6.1 (unreleased)
+- The lower boundary for packets included in ACKs is now derived, and the value sent in STOP_WAITING frames is ignored.
```
Comment (marten-seemann) [inline Changelog.md]: @lucas-clemente: We probably want to reword this. Suggestions?

**Comment by marten-seemann** [comment]
I must have closed this by mistake. Reopening.

File: Changelog.md (hunk 2 above)
Comment (lucas-clemente) [reply Changelog.md]: It's debatable whether it's worth a changelog entry, but SGTM.

File: ackhandler/interfaces.go (hunk 1 above)
Comment (lucas-clemente) [reply ackhandler/interfaces.go]: SGTM.

**Pull request merged**